src/verify/hooks.ts

```diff
@@ -27,3 +36,150 @@ export function useAsyncKomenciReadiness() {
     }
   }, [])
 }
+
+export enum PhoneNumberVerificationStatus {
+  NONE,
+  VERIFYING,
+  SUCCESSFUL,
+  FAILED,
+}
+
+export function useVerifyPhoneNumber(phoneNumber: string, countryCode: string) {
+  const verificationCodeRequested = useRef(false)
+
+  const dispatch = useDispatch()
+  const address = useSelector(walletAddressSelector)
+
+  const [verificationStatus, setVerificationStatus] = useState(PhoneNumberVerificationStatus.NONE)
+  const [verificationId, setVerificationId] = useState('')
+  const [smsCode, setSmsCode] = useState('')
+
+  const handleRequestVerificationCodeError = (error: Error) => {
+    Logger.debug(
+      `${TAG}/requestVerificationCode`,
+      'Received error from verifyPhoneNumber service',
+      error
+    )
+    dispatch(showError(ErrorMessages.PHONE_NUMBER_VERIFICATION_FAILURE))
+  }
+
+  const handleVerifySmsError = (error: Error) => {
+    Logger.debug(
+      `${TAG}/validateVerificationCode`,
+      `Received error from verifySmsCode service for verificationId: ${verificationId}`,
+      error
+    )
+    setVerificationStatus(PhoneNumberVerificationStatus.FAILED)
+    setSmsCode('')
+  }
+
+  useAsync(
+    async () => {
+      if (verificationCodeRequested.current) {
```

Comment:
Why is this needed? `useAsync` should guarantee it's only called once for a given set of passed dependencies.

Reply:
actually this is for development - the hot reload would always reload the hook and it was sending too many requests. i added a comment, do you have a better idea of how to handle this?

Reply:
oh interesting. Not sure there's a better way then.